### PR TITLE
WIP: [xmpp] Switch to xmpp.js

### PR DESCRIPTION
**Pull request by silverbucket**

### CODE REVIEW

**Comment by raucao** [comment]
That last commit added merge conflicts to the code.

**Comment by silverbucket** [comment]
> That last commit added merge conflict artifacts to the code.

@raucao Any idea how to avoid build file conflicts during merges? This has been happening a lot for me with source maps and yarn.locks

**Comment by raucao** [comment]
I usually delete the file and create a fresh one. I also usually don't merge master into a branch, but rebase on top of master, so the history is not mixed up, and the conflicts are all solved in normal history instead of special merge commits.

File: packages/sockethub-platform-xmpp/yarn.lock [hunk 1]
```diff
@@ -1233,11 +1242,14 @@ node-fetch@^2.6.1:
...
   integrity sha512-V4aYg89jEoVRxRb2fJdAg8FHvI7cEyYdVAh94HH0UIK8oJxUfkjlDQN9RbMx+bEjP7+ggMiFRprSti032Oipxw==
 
+<<<<<<< HEAD
 node-fetch@^2.6.0:
   version "2.6.1"
   resolved "https://registry.yarnpkg.com/node-fetch/-/node-fetch-2.6.1.tgz#045bd323631f76ed2e2b55573394416b639a0052"
   integrity sha512-V4aYg89jEoVRxRb2fJdAg8FHvI7cEyYdVAh94HH0UIK8oJxUfkjlDQN9RbMx+bEjP7+ggMiFRprSti032Oipxw==
 
+=======
+>>>>>>> master
```
Comment (galfert) [inline packages/sockethub-platform-xmpp/yarn.lock]: Looks like you commited a bunch of merge conflicts to the `yarn.lock` file.

Comment (raucao) [reply packages/sockethub-platform-xmpp/yarn.lock]: Looks like you didn't see the review and previous conversation in this PR. :)

Comment (galfert) [reply packages/sockethub-platform-xmpp/yarn.lock]: Whoopsie :)
Github took me straight to the changed files ¯\\_(ツ)_/¯

**Comment by silverbucket** [comment]
I don't know if I screwed something up, but I spent a while trying to update this branch as it was building up conflicts with master. Hopefully it's up to date now and I didn't loose anything.

**Comment by raucao** [comment]
Did you just merge a WIP PR? :thinking:

**Comment by silverbucket** [comment]
Man, I majorly screwed that one up, sorry. I am unsure of the state of this branch now but at least master seems to be back to normal thanks to help from @raucao ... Anyone know how I can re-open this PR? Or should I go about it another way?

**Pull request merged**